### DISCUSSION TITLE
$apply is not returning the response with OData annotations

### DISCUSSION THREAD

**Discussion by rbalagangadharan** [OData/WebApi#2840] [Q&A]
Hi All,

$apply=aggregate or groupby  returns only JSON payload in response without ODATA annotations. There is no ODATA headers in response as well.

Request:
`apply=groupby((FieldProductionsId,FieldCode),aggregate(DaysOnProduction with sum as SumOfProd))`

Response Body:
```
[
    {
        "FieldCode": 659,
        "FieldProductionsId": 2,
        "SumOfProd": 28
    }
]
```
In above the response is pure JSON not with OData annotations.

How to make the response with OData annotations, I can use the customized middleware like [this](https://github.com/peterkovecses/ODataApplyDemo), but we need to again write more logic for other OData annotations, like count, etag etc.

When we give request with  http://localhost:64771/v1/Customers?$apply=aggregate(Id with sum as TotalCount)
the outputFormatter in below screen shot is coming as `{Microsoft.AspNetCore.Mvc.Formatters.SystemTextJsonOutputFormatter}`

where as without apply `http://localhost:64771/v1/Customers?$select=Name` the outputFormatter is set to `{Microsoft.AspNetCore.OData.Formatter.ODataOutputFormatter}`.

I presume that these different makes apply to not set the OData annotations for apply. 

![image](https://github.com/OData/WebApi/assets/145420670/65bf30f4-d844-4c3c-82c7-d54451e05a7c)

![image](https://github.com/OData/WebApi/assets/145420670/3f8e1618-f28f-428d-880a-57cd380fce49)

Thanks in advance.

**Comment by habbes** [comment]
Related to https://github.com/OData/AspNetCoreOData/issues/1144
